### PR TITLE
Enable debuginfod on images

### PR DESCRIPTION
**Pull request by fao89**
https://developers.redhat.com/blog/2019/10/14/introducing-debuginfod-the-elfutils-debuginfo-server#

### CODE REVIEW

File: assets/start_script.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+debuginfod -R /var/lib/pulp -Z=cat  & /init
```
Comment (RyanSGoldberg) [inline assets/start_script.sh]:
```suggestion
debuginfod -R /var/lib/pulp -Z=cat $1 & /init
```

File: pulp/Containerfile [hunk 2]
```diff
@@ -10,6 +10,9 @@ ARG PULP_FILE_VERSION=""
 ARG PULP_MAVEN_VERSION=""
 ARG PULP_PYTHON_VERSION=""
 ARG PULP_RPM_VERSION=""
+ARG PULP_CLI_VERSION=""
+# Additional debuginfod arguments such as -d DATABASE or -t SECONDS. See man debuginfod for the full argument list
+ARG DEBUGINFOD_OPTIONS=""
```
Comment (RyanSGoldberg) [inline pulp/Containerfile:15]:
```suggestion
ARG DEBUGINFOD_OPTIONS=""
ENV DEBUGINFOD_ENV=$DEBUGINFOD_OPTIONS
```

File: pulp_ci_centos/Containerfile [hunk 3]
```diff
@@ -92,9 +93,10 @@ COPY assets/nginx.conf /nginx/nginx.conf
 COPY assets/webserver.sh /nginx/webserver.sh
 COPY assets/services /etc/services.d
 COPY assets/init /etc/cont-init.d
+COPY assets/start_script.sh /start_script.sh
```
Comment (RyanSGoldberg) [inline pulp_ci_centos/Containerfile:99]:
```suggestion
RUN chmod +x /start_script.sh
```

File: pulp_ci_centos/Containerfile [hunk 4]
```diff
@@ -92,9 +93,10 @@ COPY assets/nginx.conf /nginx/nginx.conf
...
 
 RUN /nginx/webserver.sh
 
-ENTRYPOINT ["/init"]
+ENTRYPOINT ["/start_script.sh"]
```
Comment (RyanSGoldberg) [inline pulp_ci_centos/Containerfile]:
```suggestion
# ARGs are not expanded in entry point, while ENV will be by the shell
ENV DEBUGINFOD_ENV=""
ENTRYPOINT /start_script.sh "${DEBUGINFOD_ENV}"
```

File: pulp_galaxy_ng/Containerfile [hunk 5]
```diff
@@ -2,6 +2,8 @@ ARG FROM_TAG="latest"
 FROM pulp/pulp-ci-centos:${FROM_TAG}
 
 ARG GALAXY_NG_VERSION=""
+# Additional debuginfod arguments such as -d DATABASE or -t SECONDS. See man debuginfod for the full argument list
+ARG DEBUGINFOD_OPTIONS=""
```
Comment (RyanSGoldberg) [inline pulp_galaxy_ng/Containerfile:8]:
```suggestion
ENV DEBUGINFOD_ENV=$DEBUGINFOD_OPTIONS
```

**Review by RyanSGoldberg — CHANGES_REQUESTED** [review]
This looks good on my end. I just added in a quick fix to include the DEBUGINFOD_OPTIONS in the start script

**Comment by mdellweg** [comment]
I don't know, why i am tagged for review, but let's start and ask the nasty questions:
What kind of problem is this PR solving?
How can debuginfod help us? (sounds to me that it helps with elf targets, how is it helpful for python?)
Is running debuginfod not driving us away from the goal of making the OCI images ready for production?

**Comment by fche** [comment]
> What kind of problem is this PR solving? 

It lets pulp operators share the debuginfo/debugsource content pulp downloaded with nearby users, so they don't have to do sudo yum etc. operations.

> How can debuginfod help us? (sounds to me that it helps with elf targets, how is it helpful for python?) 

It's neutral for pulp installations that redistribute python packages/rpms, except that debugging those python packages that carry native code shared libraries (and the python VM itself) become more easily debuggable.

> Is running debuginfod not driving us away from the goal of making the OCI images ready for production?

In what way?

**Pull request closed without merging**